extensions/mssql/src/objectManagement/ui/databaseDialog.ts

```diff
@@ -207,6 +243,20 @@ export class DatabaseDialog extends ObjectManagementDialogBase<Database, Databas
 		if (collationNames?.length > 0 && !collationNames.some(name => name.toLowerCase() === this.objectInfo.collationName?.toLowerCase())) {
 			errors.push(localizedConstants.CollationNotValidError(this.objectInfo.collationName ?? ''));
 		}
+
+		// Validate Rows Filegroup names
+		if (this.objectInfo.filegroups?.length > 0) {
+			let seenFilegroups = new Set(this.objectInfo.filegroups.map(function (item) { return item.name }));
+			if (seenFilegroups.size !== this.objectInfo.filegroups.length) {
```

Comment:
Since the Set is initialized using the values from `this.objectInfo.filegroups`, when do we trigger this if-statement?

Reply:
The if-statement gets triggered for every action we do on the properties dialog. As we know Filegroups will not be empty anytime, then this condition is always true and only thing we do here is prepare seenFilegroup set every time, whcih is required.

either we can do here or add a global var, but need to update the set for add/remove/edit.

Reply:
And I feel the second case is easy as we do update objectInfo for any change, and updating the seenSet will reduce multiple Set creations.